designs/sam_build_cmd.rst

```diff
@@ -0,0 +1,442 @@
+
+#. Search for a supported dependency manifest file. If a known manifest is not present, we will abort the build.
+#. Setup: Create build folder
+#. Resolve: Install dependencies
+#. Compile: Optionally, compile the code if necessary
+#. Copy Source: Optionally, Copy Lambda function code to the build folder
+
+Setup step is shared among all runtimes. Other steps in the workflow are implemented differently for each runtime.
+
+Javascript using NPM
+^^^^^^^^^^^^^^^^^^^^
+
+Install dependencies specified by ``package.json`` and copy source files
+
+**Manifest Name**: ``package.json``
+
++-------------+--------------------------------------+
+| Action      | Command                              |
++=============+======================================+
+| Resolve     | ``npm install``                      |
++-------------+--------------------------------------+
+| Compile     | No Op                                |
++-------------+--------------------------------------+
+| Copy Source | Copy files and exclude node_modules  |
++-------------+--------------------------------------+
+
+
+Java using Maven
+^^^^^^^^^^^^^^^^
+
+Let Maven take care of everything
+
+**Manifest Name**: ``pom.xml``
+
++-------------+-----------------+
+| Action      | Command         |
++=============+=================+
+| Resolve     | No Op           |
++-------------+-----------------+
+| Compile     | ``mvn package`` |
++-------------+-----------------+
+| Copy Source | No Op           |
++-------------+-----------------+
+
+Golang using Go CLI
+^^^^^^^^^^^^^^^^^^^
+
+Go's CLI will build the binary.
+
+**Manifest Name**: ``Gopkg.toml``
+
++-------------+---------------------------------------------------+
+| Action      | Command                                           |
++=============+===================================================+
+| Resolve     | ``dep ensure -v``                                 |
++-------------+---------------------------------------------------+
+| Compile     | ``GOOS=linux go build  -ldflags="-s -w" main.go`` |
++-------------+---------------------------------------------------+
+| Copy Source | No Op                                             |
++-------------+---------------------------------------------------+
+
+
+Dotnet using Dotnet CLI
+^^^^^^^^^^^^^^^^^^^^^^^
+
+**Manifest Name**: ``*.csproj``
+
++-------------+----------------------------------------------------------------------------------------+
+| Action      | Command                                                                                |
++=============+========================================================================================+
+| Resolve     | No Op                                                                                  |
++-------------+----------------------------------------------------------------------------------------+
+| Compile     | ``dotnet lambda package --configuration release --output-package $BUILD/package.zip``  |
```

Comment:
Same statement as the go but for .Net. Can I build for debugging locally, so I do not have to change my flow?